src/wfdrives.c

```diff
@@ -638,8 +638,11 @@ DrivesSetDrive(
    //
    if (hwndDir = HasDirWindow(hwndChild)) {
 
+     WCHAR szFileSpec[MAXPATHLEN];
+     
      AddBackslash(szPath);
-     SendMessage(hwndDir, FS_GETFILESPEC, MAXFILENAMELEN, (LPARAM)(szPath + lstrlen(szPath)));
+     SendMessage(hwndDir, FS_GETFILESPEC, COUNTOF(szFileSpec), (LPARAM)szFileSpec);
+     lstrcat(szPath, szFileSpec);

```

Comment:
Isn't this also another MAXPATHLEN being concatenated to a MAXPATHLEN buffer?

Reply:
In this very case we could keep it, because it is local to DrivesSetDrive.

But you can not explain someone why there is a MAXPATHLEN * 2 allocation without knowledge of the inner workings of FS_GETFILESPEC ( It is also not documented )

So to also keep the principle of locality and readability this was changed too.